### ISSUE TITLE
Codesigning failure on MacOS 

### ISSUE THREAD

**Issue by girishw** [firebase/firebase-ios-sdk#12695]
### Description

I have an app with the Firebase SDKs that compiles and works fine for an iOS destination. However, I now have a new MacOS destination and the build fails for it with the following error message: 

....../...app: code object is not signed at all
In subcomponent: ....../Build/Products/Beta/..../Contents/Frameworks/GoogleAppMeasurement.framework
Command CodeSign failed with a nonzero exit code

When I filter the MacOS build log for signing,  I see the attached
![Screenshot 2024-04-03 at 4 15 29 PM](https://github.com/firebase/firebase-ios-sdk/assets/32910700/b2116c81-9bf9-451c-b457-eb0d77fc6548)

:  the frameworks are not getting signed. 
When I similarly filter for the iOS build, I see that they are getting signed. 

### Firebase SDK Version

10.23.1

### Xcode Version

15.3

### Installation Method

Swift Package Manager

### Firebase Product(s)

Analytics

### Targeted Platforms

macOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by ncooke3** [comment]
Thanks for the report. Is this issue occurring on 10.23**.0** as well?

**Comment by girishw** [comment]
> Thanks for the report. Is this issue occurring on 10.23**.0** as well?

No, this does not occur on 10.23.0. Please note that I am using the new Xcode multiplatform target feature.

**Comment by kevinanderson7** [comment]
We are seeing a similar issue with Mac Catalyst after updating to version 10.23.1 from 10.23.0. An iOS destination seems to work fine.

**Comment by SwiftNativeDeveloper** [comment]
This is also happening to my team for 10.23.1.

Noteworthy, it can also happen for another third party dependency that we use that is also a swift package .binaryTarget.

**Comment by ncooke3** [comment]
Thanks, everyone. I suspect this issue will be fixed by the 10.24.0 release next week. The next release will contain signed SDKs and SDKs with a fixed structure for macOS/macCatalyst. I will keep this issue open until then.

**Comment by ncooke3** [comment]
10.24.0 is now released and all XCFrameworks have code signatures.
